plugins/services/src/js/structs/Framework.js

```diff
@@ -93,41 +93,13 @@ module.exports = class Framework extends Application {
   }
 
   getResources() {
-    // TODO: Circular reference workaround DCOS_OSS-783
-    const MesosStateStore = require("#SRC/js/stores/MesosStateStore");
-
-    const tasks = MesosStateStore.getTasksByService(this) || [];
-
-    const instances = this.getInstancesCount();
-    const {
-      cpus = 0,
-      mem = 0,
-      gpus = 0,
-      disk = 0
-    } = this.getSpec().getResources();
-
-    const frameworkResources = {
-      cpus: cpus * instances,
-      mem: mem * instances,
-      gpus: gpus * instances,
-      disk: disk * instances
-    };
-
-    // Aggregate all the child tasks resources
-    // resources of child frameworks won't be aggregated
-    return tasks
-      .filter(function(task) {
-        return task.state === "TASK_RUNNING" && !task.isStartedByMarathon;
-      })
-      .reduce(function(memo, task) {
-        const { cpus = 0, mem = 0, gpus = 0, disk = 0 } = task.resources;
-
-        return {
-          cpus: memo.cpus + cpus,
```

Comment:
Why exactly can we remove all this computation?

Reply:
Mesos does it for us and in a proper way. Back then, when I was writing this function, I didn't know that mesos gives `used_resources` info on the framework level. Since we aggregate resources for Marathon it was the way I fixed the issue. Now this aggregation is not needed anymore.

Reply:
Thank you for explaining this so thoroughly